### PR TITLE
Add skeleton screen

### PR DESCRIPTION
**Pull request by manuthecoder**
See #6555

### CODE REVIEW

File: config/environments/development.rb [hunk 1]
```diff
@@ -83,7 +83,7 @@
 
   # Use an evented file watcher to asynchronously detect changes in source code,
   # routes, locales, etc. This feature depends on the listen gem.
-  config.file_watcher = ActiveSupport::EventedFileUpdateChecker
+  # config.file_watcher = ActiveSupport::EventedFileUpdateChecker
```
Comment (malted) [inline config/environments/development.rb]: Did you mean to commit this?

**Comment by malted** [comment]
Hey! Would you mind adding screenshots & describing what the PR does?

File: config/environments/development.rb (hunk 1 above)
Comment (manu-gurudath) [reply config/environments/development.rb]: oops - I was having some issues when updating my files locally. I'll uncomment this line - thank you for letting me know!

**Comment by manu-gurudath** [comment]
> Hey! Would you mind adding screenshots & describing what the PR does?

Here's a brief video of what the PR does:

https://github.com/hackclub/hcb/assets/77016441/01fb8bd9-23bc-4a8e-bd80-34f6ec31f5e4


Essentially, this PR is a rewrite of the old loading animation for transaction popovers

File: app/assets/stylesheets/components/_shimmer.scss [hunk 2]
```diff
@@ -1,89 +1,81 @@
...
+  animation: shine 1s infinite;
+}
+
+html[data-dark='true'] {
+  .shimmer__rectangle::after,
+  .shimmer__circle::after {
+    background: linear-gradient(
+      to right,
+      transparent,
+      rgba(200, 200, 200, 0.1),
```
Comment (malted) [inline app/assets/stylesheets/components/_shimmer.scss]: Is it possible to use a palette colour here & in other places or does it mess up the gradient?

File: app/views/canonical_transactions/_canonical_transaction.html.erb [hunk 3]
```diff
@@ -158,65 +158,7 @@
...
-                    -
-                    <%= inline_icon "external", size: 24, class: "muted", 'aria-label': "Icon indicating click for more" %>
-                  </span>
-                </action>
-              </p>
-            </section>
-          </article>
-        </div>
-      </div>
+      <%= render "canonical_pending_transactions/skeleton" %>
```
Comment (garyhtou) [inline app/views/canonical_transactions/_canonical_transaction.html.erb]:
```suggestion
      <%= render "hcb_codes/skeleton" %>
```

**Comment by garyhtou** [comment]
This is largely to good 🪄. @sampoder can you review (test to make sure it works) and merge?

**Comment by sampoder** [comment]
Yes, will do today.

**Review by sampoder — CHANGES_REQUESTED** [review]
Hey Manu! Thanks for making this PR. At the moment, I think that this PR needs a little tweaking before we can merge this.

My main point of feedback is that it was helpful to have the transaction card in the previous version,  would it be possible to integrate a skeleton into this card:

Screenshot 2024-06-13 at 9 49 31 AM

See in the screenshot where there are dashes, those could be skeleton-ized. My second piece of feedback is that we should handle receipts. We have a rough count of the amount of receipts. Could we add a skeleton for those?

Thanks so much!

**Comment by manu-gurudath** [comment]
Hi! I've gone ahead and increased the detail within the skeleton card as seen here:

image

Up to 4 reciept skeleton loaders are now visible, as seen here:  
image

**Comment by sampoder** [comment]
I think that that card still doesn't look like the card we end up rendering? It has four columns whilst ours only has one.

**Comment by sampoder** [comment]
At the moment, I still think the current skeleton is closer to what we want and that it's styles look more like HCB's. Could you work on using the original styles and using that to build skeletons for comments and receipts?

**Comment by manu-gurudath** [comment]
@sampoder What are your thoughts on having a basic spinner instead, using pure CSS? This would simplify the codebase, and would look nice in my opinion. 

See a demo: https://jsfiddle.net/tg218uwc/5/

**Comment by sampoder** [comment]
I feel like I'd still prefer what we already have? I'm not exactly sure what's wrong with it

**Comment by manu-gurudath** [comment]
I was suggesting this because we have more than just _transaction_ popovers - there's popovers for cards and transfers as well. Using a spinner could make it easier, but I could build a skeleton UI for those in the future if you'd like

**Comment by sampoder** [comment]
Yeah I agree with some sort of general purpose loading screen. Maybe not a spinner but some sort of full screen shimmer?

**Comment by garyhtou** [comment]
Closing (on call with @manuthecoder)

**Pull request closed without merging**